subiquity/models/filesystem.py

```diff
@@ -118,7 +118,7 @@ def asdict(inst):
 @attr.s
 class Disk:
 
-    id = attr.ib(default=id_factory("disk"))
+    id = attr.ib(default=id_factory("disk"), cmp=False)
```

Comment:
In some branch or other I actually set cmp=False on the whole class. I guess some tests will need to change if we do that.

Reply:
I don't think we need to cmp=False the whole class; the two changing items are the always incrementing id value; which isn't really needed in the Disk class itself, and the private _info which is a copy of the probe data in StorageInfo form.

Reply:
We really want identity semantics for these objects I think. Adding cmp=False to the id flies in the face of that. For example, if you have two partitions of the same size on the same disk, they compare equal if you add cmp=False to id! That means if the user selects one at to delete and we exectute model._partitions.remove(part), it will always be the first one that gets removed.

So I'm going to be obstinate here I'm afraid and say we'll have to make the tests more awkward (a helper that call asdict on both things, throws out the ids and compares the dicts?) rather than add these cmp=False flags.

Reply:
Clearly two partitions of the same size on the same disk should not compare equal; they have other attributes besides id that are different.  What's missing is _eq_ vs _cmp_;  I believe I was dealing with this before switching to attr ; that is partitions can be equivalent, but not _the_ same object.  It's not clear to me that dropping id from the comparison results in having two partitions of the same size being considered the same object.

I'll dig into this a bit more before we merge the branch.